events_mgmt/src/main/java/com/adobe/event/management/ProviderService.java

```diff
@@ -41,8 +42,23 @@
 
   Optional<Provider> createProvider(final ProviderInputModel providerCreateModel);
 
+  /**
+   *
+   * @param providerInputModel the input payload
+   * @return create and if conflict/409 arises, instead, update a provider using the provided payload
+   */
+  Optional<Provider> createOrUpdateProvider(ProviderInputModel providerInputModel);
+
   Optional<Provider> updateProvider(final String id, final ProviderInputModel providerUpdateModel);
 
+  Optional<Provider> findCustomEventsProviderByInstanceId(String instanceId);
+
+  /**
+   *
+   * @param providerMetadataId
+   * @param instanceId
+   * @return the providers list matching the provided criteria and with non-empty event metadata list
+   */
```

Comment:
Javadoc needs to be above the method `findCustomEventsProviderByInstanceId`. :)

Reply:
I added it